xds/src/test/java/io/grpc/xds/SharedCallCounterMapTest.java

```diff
@@ -62,4 +62,22 @@ public boolean isDone() {
     map.cleanQueue();
     assertThat(counters).isEmpty();
   }
+
+  @Test
+  public void gcAndRecreate() {
+    @SuppressWarnings("UnusedVariable") // assign to null for GC only
+    AtomicLong counter = map.getOrCreate(CLUSTER, EDS_SERVICE_NAME);
+    final CounterReference ref = counters.get(CLUSTER).get(EDS_SERVICE_NAME);
```

Comment:
FYI, you can call `ref.clear()` and `ref.enqueue()` manually instead of relying on GC.